controllers/remote/cluster_cache_tracker_test.go

```diff
@@ -203,11 +208,19 @@ var _ = Describe("ClusterCache Tracker suite", func() {
 		Context("when Watch is called for a second time with the same input", func() {
 			BeforeEach(func() {
 				By("Calling watch on the test cluster")
+
+				kind := &corev1.Node{}
+				eventHandler := &handler.EnqueueRequestsFromMapFunc{ToRequests: handler.ToRequestsFunc(mapper)}
+
+				// Check the second copies match
+				Expect(kind).To(Equal(input.kind))
+				Expect(fmt.Sprintf("%#v", eventHandler)).To(Equal(fmt.Sprintf("%#v", input.eventHandler)))
+
 				Expect(cct.Watch(ctx, WatchInput{
 					Cluster:      input.clusterKey,
 					Watcher:      input.watcher,
-					Kind:         input.kind,
-					EventHandler: input.eventHandler,
+					Kind:         kind,
+					EventHandler: eventHandler,
 					Predicates:   input.predicates,
 				})).To(Succeed())
```

Comment:
Should we also assert that the number of watches remains stable?

Reply:
In the assertWatch a couple of lines below it asserts that there is the expected number of watches, which in this case will be 1, so we are already doing that

Reply:
github diff was hiding it, thanks!